src/server/services/wfs3/qgswfs3handlers.cpp

```diff
@@ -1390,19 +1567,331 @@ void QgsWfs3CollectionsFeatureHandler::handleRequest( const QgsServerApiContext
+      }
+
+#endif
+
+      QgsChangedAttributesMap changedAttributes;
+      QgsAttributeMap changedMap;
+      const QgsGeometryMap changedGeometries;  // This will be empty
+
+      try
+      {
+        // Parse
+        json postData = json::parse( context.request()->data() );
+
+        // If the request contains "add" we raise
+        if ( postData.contains( "add" ) )
+        {
+          throw QgsServerApiNotImplementedException( QStringLiteral( "\"add\" instruction in PATCH method is not implemented" ),
+              QString::fromStdString( QgsServerOgcApi::mimeType( contentTypeFromRequest( context.request() ) ) ),
+              400 );
+        }
+
+        // If the request does NOT contain "modify" we raise
+        if ( ! postData.contains( "modify" ) )
+        {
+          throw QgsServerApiBadRequestException( QStringLiteral( "Missing \"modify\" instruction in PATCH method" ) );
+        }
+
+        // Process attributes
+        try
+        {
+          const QgsFields authorizedFields { publishedFields( mapLayer, context ) };
+          QStringList authorizedFieldNames;
+          for ( const auto &f : authorizedFields )
+          {
+            authorizedFieldNames.push_back( f.name() );
+          }
+          const QVariantMap properties { QgsJsonUtils::parseJson( postData["modify"].dump( ) ).toMap( ) };
+          const QgsFields fields = mapLayer->fields();
+          int fieldIndex = 0;
+          for ( const auto &field : fields )
+          {
+            if ( ! properties.value( field.name() ).isNull() )
+            {
+              if ( ! authorizedFieldNames.contains( field.name() ) )
+              {
+                throw QgsServerApiPermissionDeniedException( QStringLiteral( "Feature field '%1' change is not allowed" ).arg( field.name() ) );
+              }
+              else
+              {
+                QVariant value { properties.value( field.name() ) };
+                // Convert blobs
+                if ( ! properties.value( field.name() ).isNull() && static_cast<QMetaType::Type>( field.type() ) == QMetaType::QByteArray )
+                {
+                  value = QByteArray::fromBase64( value.toByteArray() );
+                }
+                changedMap.insert( fieldIndex, value );
+              }
+            }
+            else
+            {
+              // Do nothing
+            }
+            fieldIndex++;
+          }
+          if ( ! changedMap.isEmpty() )
+          {
+            changedAttributes.insert( feature.id(), changedMap );
+          }
+        }
+        catch ( json::exception & )
+        {
+          throw QgsServerApiBadRequestException( QStringLiteral( "Feature properties are not valid" ) );
+        }
+
+      }
+      catch ( json::exception & )
+      {
+        throw QgsServerApiBadRequestException( QStringLiteral( "Feature properties are not valid" ) );
+      }
+
+      // TODO: raise if nothing to change?
```

Comment:
I'm not convinced about raising an exception in this case, but a log message would be welcome I think :).